File: src/DataTypes/MyStack.java
package DataTypes;

import java.util.ArrayList;
import java.util.List;
import java.util.Stack;
import Exception.EmptyMyStackException;

/**
 * Created by vladc on 22.10.2016.
 */
public class MyStack<T> implements MyIStack<T> {
    private Stack<T> stack;

    public MyStack() {
        stack = new Stack<>();
    }

    public T pop() throws EmptyMyStackException {
        if (stack.isEmpty())
            throw new EmptyMyStackException("Exception. MyStack is empty.");
        return stack.pop();
    }

    public void push(T value) {
        stack.push(value);
    }

    public boolean isEmpty() {
        return stack.empty();
    }

    @Override
    public String toString() {
        List<T> arr = new ArrayList<>(this.stack);
        String str = "";

        for (int i = arr.size() - 1; i > -1; i--) {
            str += arr.get(i).toString();
            str += "\n";
        }
        return str;
    }
}

File: Toy-Language-Interpreter/DataTypes/MyStack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Toy_Language_Interpreter.MyException;

namespace Toy_Language_Interpreter.DataTypes
{
    class MyStack<T> : MyIStack<T>
    {
        private Stack<T> stack;

        public MyStack()
        {
            stack = new Stack<T>();
        }

        public T Pop()
        {
            if (stack.Count == 0)
                throw new EmptyMyStackException("Exception. MyStack is empty.");
            return stack.Pop();
        }

        public void Push(T value)
        {
            stack.Push(value);
        }

        public bool IsEmpty()
        {
            return stack.Count == 0;
        }

        public override string ToString()
        {
            T[] arr = stack.ToArray();
            string str = "";

            for (int i = arr.Length - 1; i > -1; i--)
            {
                str += arr[i].ToString();
                str += "\n";
            }
            return str;
        }
    }
}
